java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -588,7 +588,35 @@ void testReadParquetBuffer() throws IOException {
 
   @Test
   void testReadParquetFull() {
-    try (Table table = Table.readParquet(TEST_PARQUET_FILE)) {
+    ParquetOptions opts = ParquetOptions.builder()
+            .includeColumn("loan_id")
+            .includeColumn("orig_channel")
+            .includeColumn("orig_interest_rate")
+            .includeColumn("orig_upb")
+            .includeColumn("orig_loan_term")
+            .includeColumn("orig_date")
+            .includeColumn("first_pay_date")
+            .includeColumn("orig_ltv")
+            .includeColumn("orig_cltv")
+            .includeColumn("num_borrowers")
+            .includeColumn("dti")
+            .includeColumn("borrower_credit_score")
+            .includeColumn("first_home_buyer")
+            .includeColumn("loan_purpose")
+            .includeColumn("property_type")
+            .includeColumn("num_units")
+            .includeColumn("occupancy_status")
+            .includeColumn("property_state")
+            .includeColumn("zip")
+            .includeColumn("mortgage_insurance_percent")
+            .includeColumn("product_type")
+            .includeColumn("coborrow_credit_score")
+            .includeColumn("mortgage_insurance_type")
+            .includeColumn("relocation_mortgage_indicator")
+            .includeColumn("quarter")
+            .includeColumn("seller_id")
```

Comment:
I'm not sure how to achieve this using `includeColumn`, but having `includeColumns` that takes an array of names allows this behavior - skip the call to `includeColumns` if you want to read all columns. If you want to read none, pass an empty array.

Reply:
Ideally this change should not have a significant impact on the API.  If no columns are specified (i.e.:  no `includeColumn`  or `includeColumns` calls) then it should read all columns as it did before.  That means all the existing tests should run without modification.  However the code explicitly throws if no columns were ever specified, which required many existing tests to be updated.  That's the thing we need to fix, otherwise there's no way to load a Parquet file without knowing the columns up-front.

> having includeColumns that takes an array of names allows this behavior - skip the call to includeColumns if you want to read all columns. If you want to read none, pass an empty array.

Or don't call `includeColumn` or `includeColumns` at all, which is the way it used to work.  If we want to add special behavior for a case where the caller _explicitly_ does not want to load any columns (e.g.: `opts.includeColumns(new String[0])` then we can do so.  However I'm not sure we can actually load an empty `Table`.  `Table` wraps a `cudf::table_view` and the latter does not allow construction with zero columns.  It seems like the Java change, if any, should be checking for no columns specified when `includeColumns` is called and throwing there rather than checking for no columns when trying to load the Parquet file, as the latter means we want to load all columns rather than no columns.  If that's done then no existing tests would be modified, all we would do is add a new test that verifies the API throws in that corner case.